registry/proxy/proxyblobstore.go

```diff
@@ -130,31 +101,37 @@ func (pbs *proxyBlobStore) ServeBlob(ctx context.Context, w http.ResponseWriter,
 	inflight[dgst] = struct{}{}
 	mu.Unlock()
 
-	// storeLocalCtx will be independent with ctx, because ctx it used to fetch remote image.
-	// There would be a situation, that is pulling remote bytes ends before pbs.storeLocal( 'Copy', 'Commit' ...)
-	// Then the registry fails to cache the layer, even though the layer had been served to client.
-	storeLocalCtx, cancel := context.WithCancel(context.Background())
-	go func(dgst digest.Digest) {
-		defer cancel()
-		if err := pbs.storeLocal(storeLocalCtx, dgst); err != nil {
-			dcontext.GetLogger(storeLocalCtx).Errorf("Error committing to storage: %s", err.Error())
-		}
-
-		blobRef, err := reference.WithDigest(pbs.repositoryName, dgst)
-		if err != nil {
-			dcontext.GetLogger(storeLocalCtx).Errorf("Error creating reference: %s", err)
-			return
-		}
-
-		pbs.scheduler.AddBlob(blobRef, repositoryTTL)
-	}(dgst)
-
-	_, err = pbs.copyContent(ctx, dgst, w)
+	defer func() {
+		mu.Lock()
+		delete(inflight, dgst)
+		mu.Unlock()
+	}()
+
+	desc, err := pbs.remoteStore.Stat(ctx, dgst)
```

Comment:
please move the Stat to line 114, just before Create(), to avoid unnecessy load for failure case.

Reply:
and please wrapper an method for the code block.

All the code should be in the same level, 
```

func {
  method()
  method()
  details....
}

== >

func {
  method()
  method()
  method()
}
```